include/fuse_lowlevel.h

```diff
@@ -2050,8 +2051,14 @@ struct fuse_session *fuse_session_new(struct fuse_args *args,
  * @return -errno  if failed to allocate memory to store `io`
  *
  **/
+#if FUSE_USE_VERSION < FUSE_MAKE_VERSION(3, 17)
 int fuse_session_custom_io(struct fuse_session *se,
-				   const struct fuse_custom_io *io, int fd);
+			   const struct fuse_custom_io *io, int fd);
+#else
+int fuse_session_custom_io_317(struct fuse_session *se,
+			       const struct fuse_custom_io *io, size_t io_size, int fd);
+#define fuse_session_custom_io(_SE, _IO, _FD) fuse_session_custom_io_317(_SE, _IO, (sizeof(_IO)), _FD)
```

Comment:
This makes it API compatible, but redirects with macro to the fixed function name _317. To make it ABI compatible with versioned symbols, like this
```
int fuse_session_loop_mt_312(struct fuse_session *se, struct fuse_loop_config *config);
FUSE_SYMVER("fuse_session_loop_mt_312", "fuse_session_loop_mt@@FUSE_3.12")
int fuse_session_loop_mt_312(struct fuse_session *se, struct fuse_loop_config *config)
````

Reply:
I think this is probably correct as-is but I agree this should use versioned symbols, I'm still trying to get my head around them completely.

Reply:
Are you sure?  The FUSE_SYMVER macro is not available to the public headers.

Reply:
This is a libfuse internal file, so it doesn't need to be in public headers. The public header basically has to define the API for 

```
#if LIBFUSE_BUILT_WITH_VERSIONED_SYMBOLS
   #if FUSE_USE_VERSION < FUSE_MAKE_VERSION(3,17)
       int fuse_session_custom_io(struct fuse_session *se, const struct fuse_custom_io *io,
			   int fd);
  #else
       int fuse_session_custom_io(struct fuse_session *se, const struct fuse_custom_io *io,
			   size_t fuse_custom_io_sz, int fd);
  #endif
#else
   /* macro solution as you already did */
#endif
```

Reply:
This is the public header.

Reply:
Ah now I see what you mean, FUSE_SYMVER() needs to be done inside of the c file, as with the other compat symbols.